packages/client/hmi-client/src/components/responsive-matrix/cell-bar-container.vue

```diff
@@ -0,0 +1,153 @@
+<template>
+	<ResponsiveCellBarGraph
+		class="cell cell-selected"
+		:style="getSelectedCellStyle(selectedCell)"
+		:update="update"
+		:selectedCell="selectedCell"
+		:dataRowList="dataRowList"
+		:dataColList="dataColList"
+		:labelRowList="labelRowList"
+		:labelColList="labelColList"
+		:parameters="parameters"
+		:parametersMin="parametersMin"
+		:parametersMax="parametersMax"
+		:colorFn="colorFn"
+		@click="$emit('click')"
+	/>
+</template>
+
+<script lang="ts">
+import { PropType } from 'vue';
+
+import { SelectedCell, SelectedCellValue } from '@/types/ResponsiveMatrix';
+import ResponsiveCellBarGraph from './cell-bar-graph.vue';
+
+export default {
+	// ---------------------------------------------------------------------------- //
+	// emits                                                                        //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	emits: ['click'],
+
+	// ---------------------------------------------------------------------------- //
+	// components                                                                   //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	components: {
+		ResponsiveCellBarGraph
+	},
+
+	// ---------------------------------------------------------------------------- //
+	// props                                                                        //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	props: {
+		getSelectedCellStyle: {
+			type: null,
+			default() {
+				return { top: '0px', left: '0px', bottom: '0px', right: '0px' };
+			}
+		},
+		update: {
+			type: Number,
+			default() {
+				return 0;
+			}
+		},
+		selectedCell: {
+			type: Array as unknown as PropType<SelectedCell>,
+			default() {
+				return [0, 0, 0, 0];
+			}
+		},
+		dataRowList: {
+			type: null,
+			default() {
+				return [];
+			}
+		},
+		dataColList: {
+			type: null,
+			default() {
+				return [];
+			}
+		},
```

Comment:
You have numerous instances where you have things declared to type=null or type=any, can we address some of these?